### PR TITLE
refactor DSL i18n serialization logic

### PR DESCRIPTION
**Pull request by Hamms**
# Description

Specifically, move shared functionality up into the base class (LevelDSL) and reduce the amount of work needed in each of the individual subclasses. Also combine the `i18n_hash` and `i18n_strings` methods into just `i18n_hash`

Should make it much easier to see at a glace which fields are being translated in each DSL class, and will help make https://codedotorg.atlassian.net/browse/FND-737 easier to implement.

Will have a side effect of reordering the hashes in the i18n directory, which will create a significant diff but should have no functional effect:

```diff
[~/cdo (i18nize-all-dsl-fields)]$ git diff i18n
diff --git a/i18n/locales/source/course_content/2017/coursec-2017.json b/i18n/locales/source/course_content/2017/coursec-2017.json
index 51a16905486..dabfb684583 100644
--- a/i18n/locales/source/course_content/2017/coursec-2017.json
+++ b/i18n/locales/source/course_content/2017/coursec-2017.json
@@ -91,11 +91,6 @@
     "contained levels": [
       {
         "dsls": {
-          "questions": [
-            {
-              "text": "Read through the code below very carefully.  What will happen when you click \"Run\"?"
-            }
-          ],
           "answers": [
             {
               "text": "The bird will end up one step short of the pig.",
@@ -117,6 +112,11 @@
               "correct": false,
               "feedback": "This code gets the bird perfectly to the pig."
             }
+          ],
+          "questions": [
+            {
+              "text": "Read through the code below very carefully.  What will happen when you click \"Run\"?"
+            }
           ]
         }
       }
@@ -201,11 +201,6 @@
     "contained levels": [
       {
         "dsls": {
-          "questions": [
-            {
-              "text": "Look closely at the code below.  What is the first bug that will make this program fail? "
-            }
-          ],
           "answers": [
             {
               "text": "The program won't fail. Scrat will get to the acorn.",
@@ -227,6 +222,11 @@
               "correct": false,
               "feedback": "The first error is that Scrat goes the wrong way after his final turn. "
             }
+          ],
+          "questions": [
+            {
+              "text": "Look closely at the code below.  What is the first bug that will make this program fail? "
+            }
           ]
         }
       }
```

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-737)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/dsl/base_dsl.rb [hunk 1]
```diff
@@ -30,14 +30,9 @@ def parse_output
     @hash
   end
 
-  # after parse has been done, this function returns a hash of all the user-visible strings from this instance
+  # override in subclass
```
Comment (daynew) [inline dashboard/app/dsl/base_dsl.rb]: Would it be helpful to have docs for this field?

File: dashboard/app/dsl/content_dsl.rb [hunk 2]
```diff
@@ -38,17 +38,14 @@ def method_missing(key, *args)
...
-    strings = {}
-    %i(
-      title
+  # @override
+  def self.i18n_fields
+    super + %w(
       content1
       content2
       content3
-      pre_title
```
Comment (daynew) [inline dashboard/app/dsl/content_dsl.rb:48]: Quick double check that `pre_title` should be removed.

**Review by daynew — APPROVED** [review]
LGTM. Just some minor comments.

File: dashboard/app/dsl/content_dsl.rb (hunk 2 above)
Comment (Hamms) [reply dashboard/app/dsl/content_dsl.rb:48]: yep! See https://github.com/code-dot-org/code-dot-org/blob/82c978b50bfe24f16f4eb2d00df2db08752b7383/dashboard/app/dsl/content_dsl.rb#L30-L34

File: dashboard/app/dsl/base_dsl.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/dsl/base_dsl.rb]: oh, oops; I had a comment here, but accidentally moved it to level_dsl

**Pull request merged**